core/state/trie_prefetcher.libevm.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 the libevm authors.
```

Comment:
Should this PR be merged into the libevm branch perhaps? 🤔 Seeing a lot of libevm specific files 🤔

Reply:
this PR intends to ship the libevm code for the trie prefetcher to production to de-risk the overall refactoring, so it includes the libevm code verbatim.